src/core/border_router/routing_manager.cpp

```diff
@@ -445,8 +566,14 @@ void RoutingManager::EvaluateRoutingPolicy(void)
 
 void RoutingManager::StartRouterSolicitation(void)
 {
+    uint32_t randomDelay;
+
     mRouterSolicitCount = 0;
-    mRouterSolicitTimer.Start(Random::NonCrypto::GetUint32InRange(0, kMaxRtrSolicitationDelay * 1000));
+
+    randomDelay = Random::NonCrypto::GetUint32InRange(0, kMaxRtrSolicitationDelay * 1000);
```

Comment:
This call requires that `kMaxRtrSolicitationDelay` > 0

```suggestion
    static_assert(kMaxRtrSolicitationDelay > 0);
    randomDelay = Random::NonCrypto::GetUint32InRange(0, kMaxRtrSolicitationDelay * 1000);
```